ios/CAExample/NativeBridge.m

```diff
@@ -12,86 +12,151 @@ @implementation NativeBridge
 
 RCT_EXPORT_MODULE(NativeBridge);
 
++ (id)allocWithZone:(NSZone *)zone {
+  static NativeBridge *sharedInstance = nil;
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    sharedInstance = [super allocWithZone:zone];
+  });
+  return sharedInstance;
+}
+
+- (NSArray<NSString *> *)supportedEvents {
+  return @[
+    @"fileRetrieveFailed",
+    @"fileRetrieved",
+    @"fileListFailed",
+    @"fileListRetrieved",
+    @"userAuthAccept",
+    @"userAuthDenied",
+    @"userAuthFailed",
+    @"accountsRetrieved",
+    @"accountsRetrieveFailed"
+  ];
+}
+
 RCT_EXPORT_METHOD(initSDK)
 {
-  NSLog(@"iOS SDK inited");
-  NSLog(@"Using parameters file: %@, password: %@, applicationid: %@, contractid: %@", P12_FILENAME, P12_PASSPHRASE, APPLICATION_ID, CONTRACT_ID);
-  
+  NSLog(@"digime: iOS SDK inited");
+  NSLog(@"digime: Using parameters file: %@, password: %@, applicationid: %@, contractid: %@", P12_FILENAME, P12_PASSPHRASE, APPLICATION_ID, CONTRACT_ID);
+
   NSString *privateKeyHex = [DMECryptoUtilities privateKeyHexFromP12File:P12_FILENAME password:P12_PASSPHRASE];
+  
   if (privateKeyHex) {
+    DMEClientConfiguration *config = [DMEClientConfiguration new];
+    config.maxRetryCount = 10;
+    [DMEClient sharedClient].clientConfiguration = config;
+
     [DMEClient sharedClient].authorizationDelegate = self;
     [DMEClient sharedClient].downloadDelegate = self;
     [DMEClient sharedClient].appId = APPLICATION_ID;
     [DMEClient sharedClient].contractId = CONTRACT_ID;
     [DMEClient sharedClient].privateKeyHex = privateKeyHex;
-    [[DMEClient sharedClient] authorize];
-    
+
     return;
   }
-  NSLog(@"Unable to create private key. Please check config variables.");
+  NSLog(@"digime: Unable to create private key. Please check config variables.");
 }
 
-- (NSArray<NSString *> *)supportedEvents {
-  return @[
-           @"fileData",
-           @"userAuthAccept",
-           @"userAuthReject"
-  ];
+RCT_EXPORT_METHOD(getFileList)
+{
+  NSLog(@"digime: Getting file list");
+  [[DMEClient sharedClient] getFileList];
+}
+
+RCT_EXPORT_METHOD(getAccounts)
+{
+  NSLog(@"digime: Getting account list");
+
+  [[DMEClient sharedClient] getAccountsWithCompletion:^(CAAccounts * _Nullable accounts, NSError * _Nullable error) {
+
+    if (!error) {
+      NSError * err;
```

Comment:
Please bump asterisk over to be next to name, IE `NSError *err`. We do have internal Objective-C semantics guidelines but I realise you can't see these!